### PR TITLE
Improve the dependency resolution doc - Add better examples, rename the nearest wins rule to direct dependency wins

### PR DESCRIPTION
**Pull request by nkolev92**
Fixes https://github.com/NuGet/docs.microsoft.com-nuget/issues/2434
Fixes https://github.com/NuGet/Home/issues/5400
Fixes https://github.com/NuGet/Home/issues/9861

Unfortunately in order for me to add new examples, I had to redo all the old ones because the style would have been different. 

I did all the examples in powerpoint. I'll upload the final presentation in this PR before merging that way we have a reference for future changes and others don't have to change everything :)

### CODE REVIEW

**Review by zivkan — APPROVED** [review]

File: docs/concepts/Dependency-Resolution.md [hunk 1]
```diff
@@ -17,80 +17,92 @@ When multiple packages have the same dependency, then the same package ID can ap
...
 
-When the NuGet restore process runs prior to a build, it resolves dependencies first in memory, then writes the resulting graph to a file called `project.assets.json`. It also writes the resolved dependencies to a lock file named `packages.lock.json`, if the [lock file functionality is enabled](../consume-packages/package-references-in-project-files.md#locking-dependencies).
-The assets file is located at `MSBuildProjectExtensionsPath`, which defaults to the project's 'obj' folder. 
+You can also take advantage of floating versions, such as 2.8.\*,  to avoid modifying the project to use the latest version of a package. When using floating versions, we recommend enabling the [lock file functionality](../consume-packages/package-references-in-project-files.md#locking-dependencies) to ensure repeatability.
```
[the remark this answers is not in the document]
Comment (nkolev92) [reply docs/concepts/Dependency-Resolution.md:22]: It's documented in the linked it. It talks about the forceevaluate option.

**Comment by nkolev92** [comment]
@NuGet/nuget-client @NuGet/nuget-pm 
I'd like to merge soon, so I'd appreciate any feedback you may have. 

The longer the PR stays open the more difficult it becomes to rereview and address feedback.

**Review by JonDouglas — APPROVED** [review]
Great changes. Looks like some positive additions to rid confusion around this topic and adds four general scenarios. Well done!

**Comment by nkolev92** [comment]
[Dependency-Resolution-Images.pptx](https://github.com/NuGet/docs.microsoft.com-nuget/files/9530000/Dependency-Resolution-Images.pptx)

Adding the pptx I used for future reference :)

**Comment by aortiz-msft** [comment]
@nkolev92 - Sorry for the feedback post-merge, but I found an error in the floating versions image. Specifically, the version used by NuGet should be 6.0.2 and not 1.0.0. Would you please take a look?

**Comment by nkolev92** [comment]
Oh nice catch. 
I'll fix it.

**Comment by nkolev92** [comment]
PR out https://github.com/NuGet/docs.microsoft.com-nuget/pull/2888.

**Pull request merged**